tcclient/http.go

```diff
@@ -47,6 +48,18 @@ type CallSummary struct {
 	Attempts int
 }
 
+// utility function to create a URL object based on given data
+func setURL(connectionData *ConnectionData, route string, query url.Values) (u *url.URL, err error) {
+	u, err = url.Parse(connectionData.BaseURL + route)
```

Comment:
Are there any Go packages that can construct a URL from different parts? Like a url.Join() or something?